File: android/src/main/java/com/jvtd/flutter_bt_printer/FlutterBtPrinterPlugin.java
package com.jvtd.flutter_bt_printer;

import android.content.Context;

import com.jvtd.flutter_bt_printer.utils.LPUtil;

import io.flutter.plugin.common.MethodCall;
import io.flutter.plugin.common.MethodChannel;
import io.flutter.plugin.common.MethodChannel.MethodCallHandler;
import io.flutter.plugin.common.MethodChannel.Result;
import io.flutter.plugin.common.PluginRegistry.Registrar;

/**
 * FlutterBtPrinterPlugin
 */
public class FlutterBtPrinterPlugin implements MethodCallHandler
{
  private Context context;

  private FlutterBtPrinterPlugin(Context context)
  {
    this.context = context;
  }

  /**
   * Plugin registration.
   */
  public static void registerWith(Registrar registrar)
  {
    final MethodChannel channel = new MethodChannel(registrar.messenger(), "flutter_bt_printer");
    channel.setMethodCallHandler(new FlutterBtPrinterPlugin(registrar.context()));
  }

  @Override
  public void onMethodCall(MethodCall call, Result result)
  {
    String method = call.method;
    switch (method)
    {
      case "init":
        LPUtil.getInstance(context).init();
        result.success("success");
        break;

      case "print":
        String topTitle = call.argument("top_title");
        String bottomTitle = call.argument("bottom_title");
        String qrCode = call.argument("qr_code");
        String label1Title = call.argument("label1_title");
        String label1Value = call.argument("label1_value");
        String label2Title = call.argument("label2_title");
        String label2Value = call.argument("label2_value");
        String label3Title = call.argument("label3_title");
        String label3Value = call.argument("label3_value");
        LPUtil.getInstance(context).print(topTitle, bottomTitle, qrCode, label1Title, label1Value, label2Title, label2Value, label3Title, label3Value);
        result.success("success");
        break;

      case "quit":
        LPUtil.getInstance(context).quit();
        result.success("success");
        break;

      default:
        result.notImplemented();
        break;
    }
  }
}

File: ios/Classes/FlutterBtPrinterPlugin.m
#import "FlutterBtPrinterPlugin.h"
#import "OAPrinterManager.h"

@implementation FlutterBtPrinterPlugin
+ (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar>*)registrar {
  FlutterMethodChannel* channel = [FlutterMethodChannel
      methodChannelWithName:@"flutter_bt_printer"
            binaryMessenger:[registrar messenger]];
  FlutterBtPrinterPlugin* instance = [[FlutterBtPrinterPlugin alloc] init];
  [registrar addMethodCallDelegate:instance channel:channel];
}

- (void)handleMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result {
  
    if ([@"init" isEqualToString:call.method]) {
    //初始化
    OAPrinterManager *printManager =  [OAPrinterManager sharedInstance];
      
  }else if ([@"print" isEqualToString:call.method]){
      NSDictionary *argumentDic = call.arguments;
      NSString *topTitle = argumentDic[@"top_title"];
      NSString *bottomTitle = argumentDic[@"bottom_title"];
      NSString *qrCode = argumentDic[@"qr_code"];
      NSString *label1Title = argumentDic[@"label1_title"];
      NSString *label1Value = argumentDic[@"label1_value"];
      NSString *label2Title = argumentDic[@"label2_title"];
      NSString *label2Value = argumentDic[@"label2_value"];
      NSString *label3Title = argumentDic[@"label3_title"];
      NSString *label3Value = argumentDic[@"label3_value"];
      
      OAAssetManageMentModel *model = [[OAAssetManageMentModel alloc]init];
      model.topTitle = topTitle;
      model.bottomTitle = bottomTitle;
      model.qrCode = qrCode;
      model.label1Title = label1Title;
      model.label1Value = label1Value;
      model.label2Title = label2Title;
      model.label2Value = label2Value;
      model.label3Title = label3Title;
      model.label3Value = label3Value;
      
      //连接设备
      [[OAPrinterManager sharedInstance] connectPrinterWithSuccess:^{
          NSLog(@"连接成功");
          //开始打印
             [[OAPrinterManager sharedInstance] printWithModel:model success:^{
                 NSLog(@"打印成功");
             } failure:^{
                 NSLog(@"打印失败");
             }];
      } failure:^{
          NSLog(@"连接失败");
      }];
      
  }else if ([@"quit" isEqualToString:call.method]){
      
      
  }else {
    result(FlutterMethodNotImplemented);
  }
    
    
}

@end
